### ISSUE TITLE
Allow `main.parameters.json` to be read from `.azure/{environmentName}/`

### ISSUE THREAD

**Issue by JayDoubleu** [Azure/azure-dev#2319]
There seems to be a requirement that the parameters file such as `main.parameters.json`, must be located in the same directory as `main.bicep`.

For enhancing flexibility and organization, it would be beneficial to allow the `main.parameters.json` file to be read from environment  directory, `.azure/{environmentName}/`.

In the scenario where the `main.parameters.json` file is not found in the `.azure/{environmentName}/` directory, the system should fall back to the default behaviour of looking in the same directory as `main.bicep`.

This modification would provide more flexibility in structuring projects and managing environments.

**Comment by JayDoubleu** [comment]
@ellismg
Yes, it's a common pattern to define multiple environments and then provision each one by simply providing `--environment xxx.` While I understand that the main purpose of this project is to enable developers to swiftly provision solutions ready for demonstration, I believe it's filling a significant gap in the tooling used by platform engineering teams.

I envision this tool being primarily developed by platform engineering teams, pre-tested and ready for developers to incorporate their code into the src folder.

Personally, I would also recommend changing the .azure folder from being hidden to being a first-class configuration folder. This is often the case for bicep and terraform code we develop. We would simply instruct the end user of this code to "copy" `.azure/environment_template` into their desired environment name, change default values, and then run `azd up --environment`.

As it stands right now, we have no tool do deploy the code AND package apps with `zipdeploy` as an option. If we would only add `--plan` or `--whatIf` to this workflow It's going to gain popularity amongst platform engineering teams very quickly.

But that's just my opinion

**Comment by jongio** [comment]
Jay - Have you seen `azd env refresh`?  That would get you env portability.  Have a look and LMK if that helps.

**Comment by nahlian** [comment]
Bicep now supports user-defined data types. With this, I can expose options in the .bicepparam. Each parameter gets a nice drop down list of options that the developer can select.

However, without being able to point to a specific .bicepparam file, working with multiple environments is cumbersome. 

I understand not wanting to muck around in the .azure directory. What if the parameters file name matches the environment name? IE: dev.bicepparam 

Maybe I missing something obvious but in the FAQ, there main.parameters.json per environment. How does this help with multiple environments today, substitution syntax? That means the flow is create parameters file, update with substitution syntax, and then go add the values in the .env file?

https://learn.microsoft.com/en-us/azure/developer/azure-developer-cli/faq#what-is-an-environment-name

**Comment by jongio** [comment]
> How does this help with multiple environments today, substitution syntax? That means the flow is create parameters file, update with substitution syntax, and then go add the values in the .env file?

When you call "read from env var" in the biccepparam file it will use env vars that are in the current azd env, as azd loads the env vars from the current env before executing any command.

i.e.

1. Run azd deploy
2. azd will find the current env and load that .env into process
3. Bicepparam with "load from env var" method use will automatically find that env var that is loaded into process and use it.

Make sense?

**Comment by nahlian** [comment]
I was trying to avoid using the `readEnvironmentVariable()` as I like the tab completion from the user defined types. Appears my desired flow is an edge case. Thank you for the reply and your time.
